### PR TITLE
plugins: action: change to Hold mode before sending DO_REPOSITION

### PR DESCRIPTION
**Pull request by TSC21**
Go-To's are only processed in PX4 if in _Hold_ mode. So one needs to enforce changing to this mode first before sending the `DO_REPOSITION` Mavlink command.

### CODE REVIEW

File: src/plugins/action/action_impl.cpp [hunk 1]
```diff
@@ -389,18 +389,30 @@ void ActionImpl::goto_location_async(
...
-        command, [this, callback](MavlinkCommandSender::Result result, float) {
-            command_result_callback(result, callback);
+    // Change to Hold mode first
+    _parent->set_flight_mode_async(
+        SystemImpl::FlightMode::Hold,
+        [this, yaw_deg, latitude_deg, longitude_deg, altitude_amsl_m, callback](
+            MavlinkCommandSender::Result mode_change_result, float) {
+            Action::Result action_result = action_result_from_command_result(mode_change_result);
+
+            // If successfully changed to Hold, then send the DO_REPOSITION command
```
Comment (JonasVautherin) [inline src/plugins/action/action_impl.cpp:399]: What happens otherwise? Shouldn't there be an `else` here? :sweat_smile:

**Review by JonasVautherin — CHANGES_REQUESTED** [review]
One question I have is... should `goto_location_async` change the flight mode, or should it return an error if it is not in the right flight mode and let the user do it? Because we already have an action to go to hold mode: https://github.com/mavlink/MAVSDK-Proto/blob/main/protos/action/action.proto#L100

File: src/plugins/action/action_impl.cpp (hunk 1 above)
Comment (TSC21) [reply src/plugins/action/action_impl.cpp:399]: good point. you have a suggestion?

Comment (JonasVautherin) [reply src/plugins/action/action_impl.cpp:399]: Well yeah, you should call `callback` with the error result

Comment (TSC21) [reply src/plugins/action/action_impl.cpp:399]: done

**Comment by TSC21** [comment]
> or should it return an error if it is not in the right flight mode and let the user do it?

That seems counter-intuitive. QGC/AMC do force a mode change before sending the GoTo. So I believe this should happen here as well.

**Comment by TSC21** [comment]
@JonasVautherin sorry I accidentally deleted the branch. Follow-up here: https://github.com/mavlink/MAVSDK/pull/1553.

**Pull request closed without merging**